src/main/java/com/iexec/core/chain/ChainConfig.java

```diff
@@ -28,19 +28,19 @@
 @NoArgsConstructor
 public class ChainConfig {
 
-    @Value("${chain.id}")
+    @Value("#{blockchainAdapterService.publicChainConfig.chainId}")
```

Comment:
Don't you think the app should not boot if publicChainConfig null?

Reply:
I think so. What would be the point to boot up if the scheduler can't access the chain?